public/app/plugins/datasource/elasticsearch/configuration/ElasticDetails.tsx

```diff
@@ -0,0 +1,202 @@
+import React from 'react';
+import { DataSourceSettings, EventsWithValidation, FormField, Input, regexValidation } from '@grafana/ui';
+import { ElasticsearchOptions } from '../types';
+
+const indexPatternTypes = [
+  { name: 'No pattern', value: 'none' },
+  { name: 'Hourly', value: 'Hourly', example: '[logstash-]YYYY.MM.DD.HH' },
+  { name: 'Daily', value: 'Daily', example: '[logstash-]YYYY.MM.DD' },
+  { name: 'Weekly', value: 'Weekly', example: '[logstash-]GGGG.WW' },
+  { name: 'Monthly', value: 'Monthly', example: '[logstash-]YYYY.MM' },
+  { name: 'Yearly', value: 'Yearly', example: '[logstash-]YYYY' },
+];
+
+const esVersions = [
+  { name: '2.x', value: 2 },
+  { name: '5.x', value: 5 },
+  { name: '5.6+', value: 56 },
+  { name: '6.0+', value: 60 },
+  { name: '7.0+', value: 70 },
+];
+
+type Props = {
+  value: DataSourceSettings<ElasticsearchOptions>;
+  onChange: (value: DataSourceSettings<ElasticsearchOptions>) => void;
+};
+export const ElasticDetails = (props: Props) => {
+  const { value, onChange } = props;
+
+  return (
+    <>
+      <h3 className="page-heading">Elasticsearch details</h3>
+
+      <div className="gf-form-group">
+        <div className="gf-form-inline">
+          <div className="gf-form max-width-25">
+            <FormField
+              labelWidth={10}
+              inputWidth={15}
+              label="Index name"
+              value={value.database || ''}
+              onChange={changeHandler('database', value, onChange)}
+              placeholder={'es-index-name'}
+              required
+            />
+          </div>
+
+          <div className="gf-form width-14">
+            <FormField
+              labelWidth={10}
+              label="Pattern"
+              inputEl={
+                <select
```

Comment:
Any particular reason for not using the `Select` component?

Reply:
Just that Select is pretty complex component for a simple case. Do we want to use it everywhere instead of normal <select>?

Reply:
If it's too complex maybe we should have a SimpleSelect wrapper that looks like this

```
<SimpleSelect value={value} onChange={onChange}>
  <SimpleSelect.Option ... />
</SimpleSelect>
```

Reply:
I don't mean it in the sense of complicated API (though that may be true) more in the sense that if we want only simple select it seemed like an overkill to use the Select component. But if we want to use it for consistency I can use it here.